pulsar-functions/localrun/src/main/java/org/apache/pulsar/functions/LocalRunner.java

```diff
@@ -752,7 +757,17 @@ private TreeMap<String, FunctionArchive> getFunctions() throws IOException {
     }
 
     private TreeMap<String, Connector> getConnectors() throws IOException {
-        return ConnectorUtils.searchForConnectors(connectorsDir, narExtractionDirectory);
+        final TreeMap<String, Connector> connectors = ConnectorUtils.searchForConnectors(connectorsDir, narExtractionDirectory, connectorsCatalogueUrl);
+
+        final ConnectorsManager connectorsManager = new ConnectorsManager(connectorsDir, narExtractionDirectory, connectorsCatalogueUrl);
+        connectors.forEach((name, connector) -> {
+            try {
+                connectorsManager.loadConnector(name, connector.getConnectorDefinition().getSinkClass() == null ? ComponentType.SOURCE : ComponentType.SINK);
```

Comment:
do we really need to do it here ? (in localrun mode eagerly download everything?)